### PR TITLE
Email assistant webapp

### PR DESCRIPTION
**Pull request by elijahbenizzy**

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
Pretty much there. What we need to do:
- [ ] Resolve conflicts :/
- [ ] Fix some issues around state
- [ ] Scroll to bottom
- [ ] Add documentation
- [ ] Copy into a web-server
- [ ] Polish/clean-up examples

File: docs/examples/web-server.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+========================
+Exposing in a Web Server
+========================
+
+Burr makes it natural to integrate with a web-server such as FastAPI. See the [example in the repository](https://github.com/DAGWorks-Inc/burr/tree/main/examples/web-server)
+for the high-level concepts + some implementation tricks, and a fully-working, runnable demo:
```
Comment (skrawcz) [inline docs/examples/web-server.rst:6]: missing link?

File: examples/email-assistant/server.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
+import functools
```
Comment (skrawcz) [inline examples/email-assistant/server.py:1]:
```suggestion
"""
This module contains the FastAPI code that interacts with the Burr application.
"""
import functools
```

File: examples/web-server/README.md [hunk 3]
```diff
@@ -1,34 +1,223 @@
 # Running Burr in a web server
 
-We're working on fleshing out this demo, but we have an example to get started with.
+Burr is meant to run interactive apps. This means running it as part of a web-service that
+responds to requests, maanges state, and documents its capabilities. The interactive nature of Burr
```
Comment (skrawcz) [inline examples/web-server/README.md:4]:
```suggestion
responds to requests, manages state, and documents its capabilities. The interactive nature of Burr
```

File: examples/web-server/README.md [hunk 4]
```diff
@@ -1,34 +1,223 @@
...
+(moving in/out of programmatic control) means we want to think carefully about how to expose
+our burr applications to the web.
```
Comment (skrawcz) [inline examples/web-server/README.md:6]:
```suggestion
our Burr applications to the web.
```

File: examples/web-server/README.md [hunk 5]
```diff
@@ -1,34 +1,223 @@
...
+simple if you index your state table on the application ID and `partition_key`.
+
+For the compute layer, you can simply scale horizontally. The only tricky aspect is ensuring state synchronization
+and locking. As we cached the application object, we could potentially get into a position
+in which the state is out of sync. To solve this, you can either:
+
+1. Use a locking method (either in the database) to ensure that only one server is running a given application at any point
+2. Use sticky sessions/sharding to ensure that a given user always hits the same server
+
+Or possibly both.
```
Comment (skrawcz) [inline examples/web-server/README.md:187]:
```suggestion
1. Use a locking method (either in the database) to ensure that only one server is running a given application at any point
2. Use sticky sessions/sharding to ensure that a given user always hits the same server
3. Handle forking of state at the persistence layer with a custom implementation

Or possibly some combination of the above.
```

File: examples/web-server/README.md [hunk 6]
```diff
@@ -1,34 +1,223 @@
...
-2. Use the burr application object to handle incoming requests
-3. Return the full state result from the burr application object to the client
-4. Render that on the frontend
+While Burr does not operate at the data access layer, this can be easily handles at the  application layer.
+Any authentication system will tell you the user ID, which you can look in your DB to determine access
+to your partition key.
 
-This way you can keep the frontend simple (just render burr's state), and keep the backend simple (delegate
-loading/persistence/logic to burr). Then you can view in telemetry.
+## Wrapup
```
Comment (skrawcz) [inline examples/web-server/README.md:208]:
```suggestion
## Wrap up
```

**Review by skrawcz — APPROVED** [review]

File: examples/web-server/README.md (hunk 5 above)
Comment (elijahbenizzy) [reply examples/web-server/README.md:187]: Yeah thought about it but it feels messy/scary. Let's add it in and see what people use?

**Pull request merged**